mariagabriella/projetoguiado1.py

```diff
@@ -0,0 +1,111 @@
+from dataset_alunas import dataset
+
+def main():
+    print("\n---  Seja bem-vinda a Escola do Reprograma!  ---")
+    print("Sistema de informações de alunas")
+
+    while True:
+        cod_opcao = obter_opcao()
+
+        if cod_opcao == 1: incluir_nova_aluna()
+        elif cod_opcao == 2: consultar_lista_alunas()
+        elif cod_opcao == 3: consultar_faltas_aluna()
+        elif cod_opcao == 4: consultar_notas_aluna()
+        elif cod_opcao == 5: consultar_status_aprovacao()
+        elif cod_opcao == 6: print("Programa encerrado"); break
+
+def obter_opcao():
+    codigo_opcao = 0
+
+    while codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+        try:
+            codigo_opcao = int(input("\nEscolha uma opção:\n"
+                                    "1 - Incluir uma nova aluna\n"
+                                    "2 - Consultar lista de alunas\n"
+                                    "3 - Consultar faltas da aluna\n"
+                                    "4 - Consultar notas da aluna\n"
+                                    "5 - Consultar status de aprovação\n"
+                                    "6 - Sair do sistema\n"
+                                    "Opção: "))
+
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 5).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+
+        return codigo_opcao
+
+def incluir_nova_aluna():
+    nome = input("Digite o nome da aluna: ")
+    sobrenome = input("Digite o sobrenome da aluna: ")
+    turma = input("Digite a turma da aluna: ")
+
+    try:
+        notas = list(map(float, input("Digite as notas da aluna (separadas vírgula): ").split(",")))
+        presencas = list(map(lambda x: x.lower() == 'true', input("Digite a presença da aluna (True/False separados por vírgula): ").split(",")))
+        participacao = float(input("Digite a nota de participação da aluna: "))
+    except ValueError:
+        print("Entrada inválida. Tente novamente com dados válidos.")
+        return
+
+    if any(nota < 0 or nota > 10 for nota in notas):
+        print("Entrada inválida. As notas devem estar entre 0 e 10.")
+        return
```

Comment:
Já que fez uma adição de validação para a lista de notas, poderia acrescentar uma valiação para a nota de participação também :)

**Sugestão:**
```suggestion
    if any(nota < 0 or nota > 10 for nota in notas):
        print("Erro: As notas devem estar entre 0 e 10.")
        return

    if participacao < 0 or participacao > 10:
        print("Erro: A nota de participação deve estar entre 0 e 10.")
        return
```